release/train_tests/horovod/train_horovod_multi_node_test.py

```diff
@@ -0,0 +1,31 @@
+import json
+import os
+import time
+
+import ray
+from ray.train import Trainer
+from ray.train.examples.horovod.horovod_example import (
+    train_func as horovod_torch_train_func,
+)
+
+if __name__ == "__main__":
+    ray.init(address=os.environ.get("RAY_ADDRESS", "auto"))
+    start_time = time.time()
+
+    num_workers = 8
+    num_epochs = 10
+    trainer = Trainer("horovod", num_workers)
```

Comment:
Should we test GPU case?

Reply:
Doesn't seem necessary for this PR. Is there a particular behavior that you think should be tested and cannot be done with CI?

Reply:
Nope not in particular, just thought that between CPU and GPU we may prefer GPU! Not sure if Horovod in particular has any additional gotchas around multi-node GPU training.

Reply:
yeah unless there is a particular gotcha we want to test, I would say let's opt for the cheaper (cost-wise) test :)

Reply:
In this case, the code path we want to test will happen in any multi-node setup regardless of CPU or GPU.